src/algebra/order/hom/monoid.lean

```diff
@@ -0,0 +1,481 @@
+between ordered groups) given only a proof that multiplication is preserved,
+
+Implicit `{}` brackets are often used instead of type class `[]` brackets. This is done when the
+instances can be inferred because they are implicit arguments to the type `order_monoid_hom`. When
+they can be inferred from the type it is faster to use this method than to use type class inference.
+
+## Tags
+
+ordered monoid, ordered group, monoid with zero
+-/
+
+open function
+
+variables {F α β γ δ : Type*}
+
+infixr ` →*₀ `:25 := monoid_with_zero_hom
+
+instance [preorder α] [preorder β] [order_hom_class F α β] : has_coe_t F (α →o β) :=
+⟨λ f, { to_fun := f, monotone' := order_hom_class.mono _ }⟩
+
+/-- Copy of a `one_hom` with a new `to_fun` equal to the old one. Useful to fix definitional
+equalities. -/
+@[to_additive "Copy of a `zero_hom` with a new `to_fun` equal to the old one. Useful to fix
+definitional equalities."]
+protected def one_hom.copy [has_one α] [has_one β] (f : one_hom α β) (f' : α → β) (h : f' = f) :
+  one_hom α β :=
+{ to_fun := f',
+  map_one' := h.symm ▸ f.map_one' }
+
+/-- Copy of a `mul_hom` with a new `to_fun` equal to the old one. Useful to fix definitional
+equalities. -/
+@[to_additive "Copy of an `add_hom` with a new `to_fun` equal to the old one. Useful to fix
+definitional equalities."]
+protected def mul_hom.copy [has_mul α] [has_mul β] (f : mul_hom α β) (f' : α → β) (h : f' = f) :
+  mul_hom α β :=
+{ to_fun := f',
+  map_mul' := h.symm ▸ f.map_mul' }
+
+/-- Copy of a `monoid_hom` with a new `to_fun` equal to the old one. Useful to fix
+definitional equalities. -/
+@[to_additive "Copy of a `monoid_hom` with a new `to_fun` equal to the old one. Useful to fix
+definitional equalities."]
+protected def monoid_hom.copy [mul_one_class α] [mul_one_class β] (f : α →* β) (f' : α → β)
+  (h : f' = f) : α →* β :=
+{ ..f.to_one_hom.copy f' h, ..f.to_mul_hom.copy f' h }
+
+/-- Copy of a `monoid_hom` with a new `to_fun` equal to the old one. Useful to fix
+definitional equalities. -/
+protected def _root_.monoid_with_zero_hom.copy [mul_zero_one_class α] [mul_zero_one_class β]
+  (f : α →*₀ β) (f' : α → β) (h : f' = f) : α →* β :=
+{ ..f.to_zero_hom.copy f' h, ..f.to_monoid_hom.copy f' h }
+
+instance [mul_zero_one_class α] [mul_zero_one_class β] [monoid_with_zero_hom_class F α β] :
+  has_coe_t F (α →*₀ β) :=
+⟨λ f, { to_fun := f, map_one' := map_one f, map_zero' := map_zero f, map_mul' := map_mul f }⟩
+
+/-- Given two monoid with zero morphisms `f`, `g` to a commutative monoid, `f * g` is the monoid
+with zero morphism sending `x` to `f x * g x`. -/
+instance {M N} {hM : mul_zero_one_class M} [comm_monoid_with_zero N] : has_mul (M →*₀ N) :=
+⟨λ f g,
+  { to_fun := λ a, f a * g a,
+    map_zero' := by rw [map_zero, zero_mul],
+    ..(f * g : M →* N) }⟩
+
+lemma order_hom.ext_iff [preorder α] [preorder β] {f g : α →o β} : f = g ↔ ∀ a, f a = g a :=
+fun_like.ext_iff
```

Comment:
Can this be deleted and its usage (line 251, `to_order_hom_injective`) be replaced with `fun_like.ext_iff`?
```suggestion
```

Reply:
Ah! You want to copy over `fun_like.ext` (because of `ext` not doing typeclass search) but not `fun_like.ext_iff`?